src/ucp/core/ucp_am.c

```diff
@@ -463,31 +463,33 @@ static ucs_status_t ucp_am_contig_short(uct_pending_req_t *self)
 
 static ucs_status_t ucp_am_bcopy_single(uct_pending_req_t *self)
 {
-    ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
-    ucs_status_t status;
+    ucs_status_t status = ucp_do_am_bcopy_single(self, UCP_AM_ID_SINGLE,
+                                                 ucp_am_bcopy_pack_args_single);
+    ucp_request_t *req;
 
-    status = ucp_do_am_bcopy_single(self, UCP_AM_ID_SINGLE,
-                                    ucp_am_bcopy_pack_args_single);
-    if (status == UCS_OK) {
-        ucp_request_send_generic_dt_finish(req);
-        ucp_request_complete_send(req, UCS_OK);
+    if (ucs_unlikely(status == UCS_ERR_NO_RESOURCE)) {
+        return UCS_ERR_NO_RESOURCE;
     }
 
+    req = ucs_container_of(self, ucp_request_t, send.uct);
+    ucp_request_send_generic_dt_finish(req);
+    ucp_request_complete_send(req, status);
```

Comment:
should return UCS_OK in case of error, to remove from arbiter
same for next function